src/coreclr/jit/importer.cpp

```diff
@@ -12565,7 +12577,7 @@ GenTree* Compiler::impCastClassOrIsInstToTree(
         //
         // use the special helper that skips the cases checked by our inlined cast
         //
-        const CorInfoHelpFunc specialHelper = CORINFO_HELP_CHKCASTCLASS_SPECIAL;
+        const CorInfoHelpFunc specialHelper = isExpandingArray ? helper : CORINFO_HELP_CHKCASTCLASS_SPECIAL;
```

Comment:
```suggestion
        const CorInfoHelpFunc specialHelper = (helper == CORINFO_HELP_CHKCASTCLASS) ? CORINFO_HELP_CHKCASTCLASS_SPECIAL : helper;
```
It can avoid `isExpandingArray`, and probably easier to understand too.